test/timeouts.js

```diff
@@ -25,6 +53,67 @@ test('get after timeout', async function (t) {
   }
 })
 
+test('get after timeout with constructor', async function (t) {
+  t.plan(1)
+
+  const core = await create({ timeout: 1 })
+
+  try {
+    await core.get(0)
+    t.fail('should not get a block')
+  } catch (err) {
+    t.is(err.code, 'REQUEST_TIMEOUT')
+  }
+})
+
+test('session get after timeout', async function (t) {
+  t.plan(1)
+
+  const core = await create()
+  const session = core.session({ timeout: 1 })
+
+  try {
+    await session.get(0)
+    t.fail('should not get a block')
+  } catch (err) {
+    t.is(err.code, 'REQUEST_TIMEOUT')
+  }
+})
+
+test('session get after inherited timeout', async function (t) {
+  t.plan(1)
+
+  const core = await create({ timeout: 1 })
+  const session = core.session()
+
+  try {
+    await session.get(0)
+    t.fail('should not get a block')
+  } catch (err) {
+    t.is(err.code, 'REQUEST_TIMEOUT')
+  }
+})
+
+test('core constructor timeout but disable on get', async function (t) {
+  t.plan(1)
+
+  const core = await create({ timeout: 1 })
+
+  const get = core.get(0, { timeout: 0 })
+  setTimeout(() => core.append('hi'), 3)
+  t.alike(await get, b4a.from('hi'))
```

Comment:
I think it's good now, it first does the `get()` without `await`, later `setTimeout`, and afterwards awaits the first get

Reply:
So for example, if `{ timeout: 0 }` were `1` it should (tm lol) always fail no matter CI slowness, etc